\subsection{SNP calling}

To call SNPs we will use \texttt{samtools} and \texttt{bcftools}.
Using \texttt{samtools} we will generate our data in \textit{pileup} format which
is then accessed by \texttt{bcftools} to call SNPs.
Variant callers need sorted by position \textit{BAM} file to successfuly detect variants.
To sort our \textit{BAM} file we will use \texttt{samtools}. In linux terminal type:\\~\\
\sortbam{\mapReads\_mapped.reheaded.bam}{\mapReads\_mapped.reheaded.sorted.bam}\\

There are two choices of variant callers in \texttt{bcftools}:
\texttt{--consensus-caller} or \texttt{--multiallelic-caller}. \texttt{consensus-caller} is
older variant calling model and assumes only biallelic sites. It can miss somatic mutations.
\texttt{multiallelic-caller} does not have such an asumption and is more sensitive than \texttt{consensus-caller}.
%that the analyzed sample is biallelic and can miss somatic mutations. In contrast the \texttt{multiallelic-caller}
\texttt{multiallelic-caller} is recommended for most tasks, and we will also use it. For this tutorial, we will skip insertions/deletions: \\~\\
\samtoolssnp{\refRel}{\mapReads\_mapped.reheaded.sorted.bam}{\mapReads.bcftools\_snps.vcf}
