\subsection{SNP annotation and filtering with \texttt{snpEff} and \texttt{SnpSift}}
After generating a list of SNPs in \textit{VCF} format,
we will use a tool \texttt{snpEff} to predict whether they are causing
amino acid change and \texttt{SnpSift} to filter nonsynonymous and stop gained SNPs.
\subsubsection{Setting up and configuring \texttt{snpEff} and \texttt{SnpSift}}
\texttt{snpEff} and \texttt{SnpSift} are bundled together and are available at \url{http://snpeff.sourceforge.net}.
To download and install them, type in linux terminal:\\~\\
\texttt{cd \textasciitilde/\progDir} \\
\texttt{wget http://sourceforge.net/projects/snpeff/files/snpEff\_latest\_core.zip} \\

paste the copied location and hit \texttt{Enter}. To \texttt{unzip snpEff} type:\\~\\
\texttt{unzip snpEff\_latest\_core.zip}\\

Unfortunately we wil not be able to launch both programs from \texttt{\binDir} folder
and we can leave the software in \texttt{\progDir} directory.

\texttt{snpEff} requires databases to predict effects of SNPs.
The needed database is already downloaded for us and resides in
\texttt{\textasciitilde/\dataDir/\snpEffData}. Our task is to 
configure \texttt{snpEff} to tell it where this database can be found.

After downloading and unzipping of \texttt{snpEff} we will need to
edit \texttt{snpEff's} configuration file named \texttt{snpEff.config}
and change entry \texttt{data.dir = ./data/} to \texttt{data.dir = \textasciitilde/\dataDir/\snpEffData/}
%In linux terminal type:\\~\\
%\texttt{cd \textasciitilde/\progDir/snpEff}
%\texttt{nano snpEff.config}

%TODO aprakstÄ«t abas komandas
\texttt{SnpSift} and \texttt{snpEff} are \textit{.jar} files which means that
these are \texttt{java} applications. To run \texttt{java} applications through linux terminal,
use command \texttt{java -jar} \textit{filename.jar}. To sucessfully perfrom SNP
annotation, we need to provide \texttt{snpEff} with the command we are executing 
(\texttt{ann} - telling \texttt{snpEff} that we want to annotate SNPs), its configuration file (\texttt{-c snpEff.config}),
database we are annotating against (\texttt{GRCh38.76}) and our \textit{VCF} file. Since \texttt{snpEff} by default prints
output to terminal, we can save it using \texttt{>} operator.
To annotate called variants, type:\\~\\
\snpEff{\mapReads.bcftools\_snps.vcf}{\mapReads.bcftools\_snps.annotated.vcf}\\

In the resulting \textit{VCF} file \texttt{snpEff} adds a new entry in the \texttt{INFO} field - \texttt{ANN}.
To filter only those SNPs that are marked as nonsynonymous or stop gained we are going to
use \texttt{SnpSift}'s command \texttt{filter}. \texttt{SnpSift}'s \texttt{filter} command needs
\textit{VCF} file annotated by \texttt{snpEff} and filtering expression. Type:\\~\\
\SnpSiftFilter{\mapReads.bcftools\_snps.annotated.vcf}{\mapReads.bcftools\_snps.annotated.nonsyn\_stop.vcf}\\
