\documentclass{article}
\usepackage{verbatim}
\usepackage[a4paper]{geometry} 
%%%[a4paper, total={6in, 8in}]{geometry}
\usepackage[usenames,dvipsnames]{xcolor}
\usepackage{framed}
\usepackage{fontspec}
\usepackage{hhline}
\usepackage{hyperref}
\usepackage{listings}
\newcommand{\git}{\texttt{git}}
\newcommand{\Git}{\texttt{Git}}
\newcommand{\IonTorrent}{\textit{IonTorrent}}
\newcommand{\binDir}{binaries}
\newcommand{\fbinDir}{\textasciitile/\binDir}
\newcommand{\progDir}{programs}
\newcommand{\workDir}{ngs\_work}
\newcommand{\denovoDir}{denovo}
\newcommand{\reseqDir}{reseq}
\newcommand{\refSeq}{chr\_merged.fa}
\newcommand{\refDir}{ref}
\newcommand{\refFullDir}{\texttt{\textasciitile/\workDir/\reseqDir/ref}}
\newcommand{\refRel}{\texttt{ref/\refSeq}}
\newcommand{\denovo}{\textit{de-novo}}
\newcommand{\denovoReads}{denovo\_reads}
\newcommand{\dataDir}{data/day1}
\newcommand{\mapReads}{reseq\_reads}
\newcommand{\snpEffData}{snpEffData}
\newcommand{\bash}{\textit{bash}}
\newcommand{\pipename}{snp\_pipeline.sh}
\newcommand{\targz}{\textit{.tar.gz}}
\newcommand{\tarbz}{\textit{.tar.bz2}}
\newcommand{\gz}{\textit{.gz}}
%Reference genome indexing
\newcommand{\refindex}[1]{\texttt{tmap index -f #1}}
%map reads
\newcommand{\tmap}[3]{\texttt{tmap map3 -f #1~\textbackslash \\
  \indent -r #2~\textbackslash \\
  \indent -i fastq~\textbackslash \\
  \indent -o 1~\textbackslash \\
  \indent -s #3
  }
}
\newcommand{\tmapWithRG}[3]{\texttt{tmap map3 -f #1~\textbackslash \\
  \indent -r #2 \textbackslash~\\
  \indent -i fastq \textbackslash~\\
  \indent -R ID:SomeID \textbackslash~\\
  \indent -R SM:Sample1 \textbackslash~\\%"@RG\textbackslash tID:SomeID\textbackslash tSM:Sample1" \textbackslash \\
  \indent -o 1 \textbackslash~\\
  \indent -s #3
  }
}
\newcommand{\sortbam}[2]{\texttt{samtools sort #1 #2}}

\newcommand{\reheader}[2]{\texttt{samtools view -H #1~\textbackslash \\
  \indent | sed 's/ID:NOID/SM:SomeID/;s/SM:NOSM/SM:Sample1/'~\textbackslash \\
  \indent | samtools reheader - #1~\textbackslash \\
  \indent > #2
  }
}
\newcommand{\samtoolssnp}[3]{\texttt{samtools mpileup -uf #1~\textbackslash \\
  \indent #2~\textbackslash \\
  \indent | bcftools call --multiallelic-caller --variants-only --skip-variants indels~\textbackslash \\
%  \indent | vcfutils.pl varFilter > #3
  \indent > #3
  }
}

\newcommand{\snpEff}[2]{\texttt{java -jar \textasciitilde/\progDir/snpEff/snpEff.jar~\textbackslash \\
  \indent ann~\textbackslash \\
  \indent -c \textasciitilde/programs/snpEff/snpEff.config~\textbackslash \\
  \indent GRCh38.76~\textbackslash \\
  \indent #1~\textbackslash \\
  \indent > #2 
  }
}

\newcommand{\snpEffGC}[2]{\texttt{java -jar -Xmx3g \textasciitilde/\progDir/snpEff/snpEff.jar~\textbackslash \\
  \indent ann~\textbackslash \\
  \indent -c \textasciitilde/programs/snpEff/snpEff.config~\textbackslash \\
  \indent GRCh38.76~\textbackslash \\
  \indent #1~\textbackslash \\
  \indent > #2
  }
}
\newcommand{\SnpSiftAnnotate}[2]{\texttt{java -jar \textasciitilde/\progDir/snpEff/SnpSift.jar~\textbackslash \\
  \indent annotate~\textbackslash \\
  \indent \textasciitilde/\dataDir/snpEffData/00-All.vcf~\textbackslash \\
  \indent -id~\textbackslash \\
  \indent #1~\textbackslash \\
  \indent > #2
  }
}
%java -jar ~/programs/snpEff/SnpSift.jar annotate ~/data/day1/snpEffData/00-All.vcf -id reseq_reads.ann.vcf >reseq_reads.ann.id.vcf 

%java -jar ~/programs/snpEff/SnpSift.jar dbnsfp reseq_reads.ann.id.vcf >reseq_reads.ann.id.dbnsfp.vcf
\newcommand{\SnpSiftAnnotateDBNSFP}[2]{\texttt{java -jar \textasciitilde/\progDir/snpEff/SnpSift.jar~\textbackslash \\
  \indent dbnsfp~\textbackslash \\
  \indent #1~\textbackslash \\
  \indent > #2
  }
}

%database.local.dbnsfp = \textasciitilde/\dataDir/snpEffData/dbNSFP2.7.txt.gz

\newcommand{\SnpSiftFilter}[2]{\texttt{java -jar \textasciitilde/\progDir/snpEff/SnpSift.jar~\textbackslash \\
  \indent filter~\textbackslash \\
  \indent -f #1~\textbackslash \\
  \indent "(ANN[*].EFFECT = 'missense\_variant') || (ANN[*].EFFECT = 'stop\_gained')"~\textbackslash \\
  \indent > #2
  }
}

\newcommand{\SnpSiftExtractFields}[2]{\texttt{java -jar \textasciitilde/\progDir/snpEff/SnpSift.jar 
  \indent extractFields~\textbackslash \\
  \indent #1~\textbackslash \\
  \indent CHROM POS ID REF ALT dbNSFP\_SIFT\_pred GEN[*].GT~\textbackslash \\
  \indent > #2
  }
}
\lstset{
  language=bash,
  basicstyle=\footnotesize\ttfamily,
  showstringspaces=false,
  commentstyle=\color{ForestGreen},
  keywordstyle=\color{blue},
  stringstyle=\color{BrickRed}
%  breaklines=true
}

%\lstset{language=bash, basicstyle=\footnotesize\ttfamily,breaklines=true}
%pipeline:
%\refindex{\refRel}
%\tmap{\refRel}{\mapReads.fastq}{\mapReads\_mapped.bam}\\
%\tmapWithRG{\refRel}{\mapReads.fastq}{\mapReads\_mapped.reheaded.bam}\\
%\reheader{\mapReads\_mapped.bam}{\mapReads\_mapped.reheaded.bam}\\
%\texttt{\samtoolssnp{\refRel}{\mapReads\_mapped.reheaded.sorted.bam}{\mapReads.bcftools\_snps.vcf}}\\
%\snpEff{\mapReads.bcftools\_snps.vcf}{\mapReads.bcftools\_snps.annotated.vcf}\\
%\SnpSiftFilter{\mapReads.bcftools\_snps.annotated.vcf}{\mapReads.bcftools\_snps.annotated.nonsyn\_stop.vcf}\\
\begin{document}
  \sloppy
  \vspace*{\stretch{1.0}}
  \begin{center}
  \Large\textbf{Basics in command line handling, with the examples on NGS data analysis}\\~\\
  \large\textit{Ivars Silamikelis, COST Basic training on NGS data analysis, Riga 2015}
  \end{center}
  \vspace*{\stretch{2.0}}
%\input{title/title.tex}
  \input{linux/linux_basics.tex}
  \input{linux/env_setup.tex}
  \input{denovo/denovo_uzskate.tex}
  \input{snp_calling/snp_calling_uzskate.tex}
  \input{snp_calling/pipeline_building.tex}
\end{document}
